tests/integration/strategy_tests.cpp

```diff
@@ -618,8 +654,66 @@ TEST(ThreadSafeAllocator, Host)
 
   SUCCEED();
 }
+#endif
+
+#if defined(UMPIRE_ENABLE_DEVICE)
+TEST(ThreadSafeAllocator, DeviceStdThread)
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  auto allocator = rm.makeAllocator<umpire::strategy::ThreadSafeAllocator>(
+      "thread_safe_allocator", rm.getAllocator("DEVICE"));
+
+  constexpr int N = 16;
+  std::vector<void*> thread_allocs{N};
+  std::vector<std::thread> threads;
+
+  for (std::size_t i = 0; i < N; ++i)
+  {
+    threads.push_back(
+        std::thread([=, &allocator, &thread_allocs] { thread_allocs[i] = allocator.allocate(1024); }));
```

Comment:
I would suggest something a little more stressful.  I think the loop below may induce more things happening in parallel within the `allocator` object:

```suggestion
        std::thread([=, &allocator, &thread_allocs] {
            for ( j = 0; j < N; ++j)
            {
                thread_allocs[i] = allocator.allocate(1024);
                ASSERT_NE(thread_allocs[i]);
                allocator.deallocate(thread_allocs[i]);
                ASSERT_THROW({ allocator.deallocate(thread_allocs[i]); });
                thread_allocs[i] = allocator.allocate(1024);
        }));
```

Thoughts?

Reply:
I agree!